database/rpcdb/db_test.go

```diff
@@ -24,14 +24,14 @@ type testDatabase struct {
 	server *memdb.Database
 }
 
-func setupDB(t testing.TB) *testDatabase {
+func setupDB(t testing.TB, ctx context.Context) *testDatabase {
 	require := require.New(t)
 
 	db := &testDatabase{
 		server: memdb.New(),
 	}
 
-	listener, err := grpcutils.NewListener()
+	listener, err := grpcutils.NewListener(ctx)
```

Comment:
We only ever pass in `ctx = context.Background()`, and I don't see why we would ever care to pass in anything else.
```suggestion
	listener, err := grpcutils.NewListener(context.Background())
```